src/main/java/Echo.java

```diff
@@ -0,0 +1,131 @@
+import parser.Parser;
+import tasks.Task;
+import storage.Storage;
+import tasks.TaskList;
+import ui.Ui;
+
+import java.io.IOException;
+import java.time.format.DateTimeParseException;
+
+public class Echo {
+    private String word;
+    private TaskList list;
+    private final Storage storage;
+    private final Ui ui;
+    private final Parser parser;
+
+    public Echo() {
+        this.storage = new Storage("data/chatHistory.txt");
+        try {
+            this.list = storage.load();
+        } catch (IOException e) {
+            this.list = new TaskList();
+        }
+        this.ui = new Ui();
+        this.parser = new Parser();
+    }
+
+    /**
+     * Sets the word for the echo object.
+     * @param word used to initialise the word of an echo object.
+     */
+    public void setWord(String word) {
+        this.word = word;
+    }
+
+    /**
+     * Main logic for the entire chatbot. Handles all the different commands.
+     */
+    public String echoOut() {
+
+            String description = word;
+            String[] parts = description.split(" ", 2);
+            String command = parts[0];
+
+        switch (command) {
+        case "list":
+            return list.printList() + "\n" + ui.line();
+        case "mark":
+            if (parts.length < 2) {
+                return "Please specify the task number to mark \n" + ui.line();
+            } else {
+                try {
+                    int index = Integer.parseInt(parts[1]) - 1;
+                    return list.mark(index) + ui.line();
+                } catch(NumberFormatException e) {
+                    return "Please enter a valid task number\n" + ui.line();
+                }
+            }
+        case "unmark":
```

Comment:
"mark" and "unmark" have similar logic. It might be better to combine the cases, and handle them within the switch-case.